File: sort/selectionSort.js
// 首先在未排序序列中找到最小（大）元素，存放到排序序列的起始位置
// 再从剩余未排序元素中继续寻找最小（大）元素，然后放到已排序序列的末尾。
// 重复第二步，直到所有元素均排序完毕。
function selectionSort(arr) {
	var len = arr.length;
	var minIndex;
	for (var i = 0; i < len - 1; i++) {
		minIndex = i;
		// 找到最小的数
		for (var j = i + 1; j < len; j++) {
			if (arr[j] < arr[minIndex]) {
				minIndex = j;
			}
		}
		[arr[i], arr[minIndex]] = [arr[minIndex], arr[i]];
	}
	return arr;
}
console.log(selectionSort([3, 7, 1, 4, 9, 2]));

File: sort/selectionSort.cpp
#include <iostream>
#include <vector>
using namespace std;
int main() {
  /**
   * 初始化数组
   */
  int num; // 数组个数
  cin >> num;
  // vector<int> arr(num);
  // malloc动态分配内存，让开辟num 个 int大小的内存空间给arr
  // 也可以使用vector
  int *arr = (int*)malloc(num * sizeof(int));
  for (int i = 0; i < num; ++i) {
    cin >> arr[i];
    // arr.push_back(arr[i]);
  }
  /**
   * 选择排序
   */
  int minIndex;
  for (int i = 0; i < num - 1; ++i) {
    minIndex = i;
    for (int j = i + 1; j < num; ++j) {
      if (arr[j] < arr[minIndex]) {
        minIndex = j;
      }
    }
    int tmp = arr[minIndex];
    arr[minIndex] = arr[i];
    arr[i] = tmp;
  }
  for (int i = 0; i < num; ++i) {
    cout << arr[i];
  }
  cout << "" << endl;
  // C++可以用new 创建数组，如果用new 创建的，这里用delect释放内存
  free(arr);
  return 0;
}